### PR TITLE
Configuration endpoints

### PR DESCRIPTION
**Pull request by atarix83**
This connect to #183 

In addition to the base config service, I've added services for these endpoints : 

- [https://github.com/DSpace/Rest7Contract/blob/master/submissiondefinitions.md](https://github.com/DSpace/Rest7Contract/blob/master/submissiondefinitions.md)
- [https://github.com/DSpace/Rest7Contract/blob/master/submissionsections.md](https://github.com/DSpace/Rest7Contract/blob/master/submissionsections.md)
- [https://github.com/DSpace/Rest7Contract/blob/master/submissionforms.md](https://github.com/DSpace/Rest7Contract/blob/master/submissionforms.md)

To test endpoints change the rest server to `dspace7.dev01.4science.it`

### CODE REVIEW

File: src/app/core/config/config.service.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    return result;
+  }
+
+  public getConfigAll(): Observable<ConfigData> {
+    return this.getEndpoint()
+      .filter((href: string) => isNotEmpty(href))
+      .distinctUntilChanged()
+      .map((endpointURL: string) => new ConfigRequest(endpointURL))
+      .do((request: RestRequest) => {
+        setTimeout(() => {
```
Comment (artlowel) [inline src/app/core/config/config.service.ts]: These setTimeouts around `requestservice.configure()` calls are no longer necessary. They were the reason for the performance issues in described in [this comment](https://github.com/DSpace/dspace-angular/pull/180#issuecomment-339632224).

File: src/app/core/config/config.service.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  public getConfigByName(name: string): Observable<ConfigData> {
+    return this.getEndpoint()
+      .map((endpoint: string) => this.getConfigByIDHref(endpoint, name))
+      .filter((href: string) => isNotEmpty(href))
+      .distinctUntilChanged()
+      .map((endpointURL: string) => new ConfigRequest(endpointURL))
+      .do((request: RestRequest) => {
+        setTimeout(() => {
```
Comment (artlowel) [inline src/app/core/config/config.service.ts]: Same comment as above

File: src/app/core/config/config.service.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  public getConfigBySearch(options: FindAllOptions = {}): Observable<ConfigData> {
+    return this.getEndpoint()
+      .map((endpoint: string) => this.getConfigSearchHref(endpoint, options))
+      .filter((href: string) => isNotEmpty(href))
+      .distinctUntilChanged()
+      .map((endpointURL: string) => new ConfigRequest(endpointURL))
+      .do((request: RestRequest) => {
+        setTimeout(() => {
```
Comment (artlowel) [inline src/app/core/config/config.service.ts]: Same comment as above

File: src/app/core/config/config.service.ts (hunk 1 above)
Comment (atarix83) [reply src/app/core/config/config.service.ts]: I will remove

**Comment by atarix83** [comment]
> export class HomePageComponent implements OnInit {
> 
>   constructor(private cs: ConfigService) {
>   }
> 
>   ngOnInit(): void {
>     this.cs.getConfigAll().subscribe((c) => console.log('c', c));
>   }
> }

Do you use ConfigService? If so, it's only the abstract class, you should use for instance 
```
  constructor(private cs: SubmissionSectionsConfigService) {
  }
```

> Also, would you mind adding unit tests? This PR causes the coverage to go down by quite a lot.

yes, I will do it

**Comment by atarix83** [comment]
I've removed setTimeout calls and added some tests

**Comment by tdonohue** [comment]
I'm not entirely sure how to test this from the Angular UI (or if it even is testable). But, the code looks reasonable to me, and I'm glad to see specs here too.  So, it looks good to me. 👍

**Pull request merged**